lib/io/i2c.cpp

```diff
@@ -0,0 +1,77 @@
+#include "i2c.hpp"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <sys/ioctl.h>
+
```

Comment:
A lot of C headers? Can these be replaced by better C++ headers or are these better?

Reply:
ioctl has no equivalent